nautobot/extras/choices.py

```diff
@@ -304,6 +304,19 @@ class RelationshipTypeChoices(ChoiceSet):
     )
 
 
+class RelationshipSidesRequiredChoices(ChoiceSet):
+
+    NEITHER_SIDE_REQUIRED = ""
+    SOURCE_SIDE_REQUIRED = "source"
+    DESTINATION_SIDE_REQUIRED = "destination"
+
+    CHOICES = (
+        (NEITHER_SIDE_REQUIRED, "No"),
+        (SOURCE_SIDE_REQUIRED, "Source side required"),
+        (DESTINATION_SIDE_REQUIRED, "Destination side required"),
```

Comment:
Would something like this be accurate?

```suggestion
        (SOURCE_SIDE_REQUIRED, "Source objects MUST implement this relationship"),
        (DESTINATION_SIDE_REQUIRED, "Destination objects MUST implement this relationship"),
```

Just trying to think of a way to make it clearer what "required" actually means here.